### PR TITLE
Updated clean-logs.sh

### PR DESCRIPTION
**Pull request by BShraman**
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: scripts/in_container/prod/clean-logs.sh [hunk 1]
```diff
@@ -17,14 +16,14 @@
 # specific language governing permissions and limitations
 # under the License.
 
-set -euo pipefail
+set -e pipefail
```
Comment (uranusjr) [inline scripts/in_container/prod/clean-logs.sh]: Why?

File: scripts/in_container/prod/clean-logs.sh [hunk 2]
```diff
@@ -33,5 +32,5 @@ while true; do
   find "${DIRECTORY}"/logs -mtime +"${RETENTION}" -name '*.log' -delete
 
   seconds=$(( $(date -u +%s) % EVERY))
-  [[ $seconds -lt 1 ]] || sleep $((EVERY - seconds))
+  (( "${seconds}" < 1 )) || sleep $((EVERY - seconds))
```
Comment (uranusjr) [inline scripts/in_container/prod/clean-logs.sh]: Why is the suggested syntax considered better than the current one?

File: scripts/in_container/prod/clean-logs.sh (hunk 1 above)
Comment (potiuk) [reply scripts/in_container/prod/clean-logs.sh]: Yeah... why ?

Comment (potiuk) [reply scripts/in_container/prod/clean-logs.sh]: It won't work by the way :)

File: scripts/in_container/prod/clean-logs.sh (hunk 2 above)
Comment (potiuk) [reply scripts/in_container/prod/clean-logs.sh]: Actually I like the new way better. Explicit math bash expression is more readable than -lt

Comment (potiuk) [reply scripts/in_container/prod/clean-logs.sh]: However it should be simpler:

```suggestion
  (( seconds < 1 )) || sleep $((EVERY - seconds))
```

in math expression there is no need for neither quoting nor '$'

**Comment by potiuk** [comment]
And please run it through `pre-commit` before. We use `shell-check` and it should flag some of the changes you made as wrong

File: scripts/in_container/prod/clean-logs.sh (hunk 2 above)
Comment (BShraman) [reply scripts/in_container/prod/clean-logs.sh]: Suggested by google cheat-sheet for numerical comparison and for better and simpler understanding

File: scripts/in_container/prod/clean-logs.sh (hunk 1 above)
Comment (BShraman) [reply scripts/in_container/prod/clean-logs.sh]: I reverted by changes and pushed by code again. I was missing pre-commit test in my first push.

**Comment by BShraman** [comment]
Hi @potiuk , i have followed below steps before push to GIT am i missing anything ? Could you please let me know since this is my first push.
-pre-commit run --files scripts/in_container/prod/clean-logs.sh
-pre-commit install
- git commit -m "Updated clean-logs.sh"
-

**Comment by potiuk** [comment]
Should be ok. I usually also rebase my commits on top of `main` (see CONTRIBUTING - we have description about rebasing).
 This will automatically account for any changes from `main` that should be incorporated (it happens occasionally that they need to be).

**Comment by BShraman** [comment]
Sure , let me add on my steps for future push.

**Comment by BShraman** [comment]
Hi @potiuk @uranusjr anything else on this change? I wanted to be learn the process before i can start on next task.

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**